src/tcgwars/logic/impl/gen4/CallOfLegends.groovy

```diff
@@ -1151,427 +579,179 @@ public enum CallOfLegends implements LogicCardInfo {
         return basic (this, hp:HP060, type:FIGHTING, retreatCost:2) {
           weakness W
           resistance L, MINUS20
           pokeBody "Ultra-Thick Skin", {
             text "As long as Phanpy has Energy attached to it, any damage done to Phanpy by attacks is reduced by 10 ."
-            delayedA {
+            delayed {
+              before APPLY_ATTACK_DAMAGES, {
+                bg.dm().each {if(it.to==self && it.from.owner==self.owner.opposite && it.notZero && it.notNoEffect) {
+                  bc "$thisAbility -10"
+                  it.dmg-=hp(10)
+                }}
+              }
             }
           }
           move "Rock Smash", {
             text "20+ damage. Flip a coin. If heads, this attack does 20 damage plus 10 more damage."
             energyCost F, C
-            attackRequirement {}
             onAttack {
               damage 20
-              flip { damage 10 }
+              flip {
+                damage 10
+              }
             }
           }
 
         };
       case PIDGEY_67:
-        return basic (this, hp:HP050, type:COLORLESS, retreatCost:1) {
-          weakness L
-          resistance F, MINUS20
-          move "Messenger", {
-            text "Search your deck for a Pokémon, show it to your opponent, and put it into your hand. Shuffle Pidgey and all cards attached to it back into your deck."
-            energyCost C
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-          move "Glide", {
-            text "10 damage. "
-            energyCost C
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-
-        };
+        return copy(Triumphant.PIDGEY_71, this);
       case PINECO_68:
-        return basic (this, hp:HP060, type:GRASS, retreatCost:2) {
-          weakness R
-          move "Focus Energy", {
-            text "During your next turn, Pineco’s Surprise Attack attack’s base damage is 80."
-            energyCost C
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-          move "Surprise Attack", {
-            text "40 damage. Flip a coin. If tails, this attack does nothing."
-            energyCost G, C
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-
-        };
+        return copy(Undaunted.PINECO_63, this);
       case RELICANTH_69:
         return basic (this, hp:HP080, type:WATER, retreatCost:2) {
           weakness G
           move "Prehistoric Wisdom", {
             text "Choose a card from your hand and put it in the Lost Zone. Then, draw 3 cards."
             energyCost C
-            attackRequirement {}
+            attackRequirement {
+              assert my.deck||my.hand : "You have no cards in your hand or deck!"
+            }
             onAttack {
-              damage 0
+              if(my.hand) {
```

Comment:
Hand check shouldn't be required since attackRequirement handles this.

Reply:
I think the hand check is needed as a a legal play would to attack with no cards in hand and draw 3 cards without having put any into the lost zone.

Reply:
My bad. I forgot select doesn't check if the cardlist is empty before building the UI. I'll say this is good for now.

However, @axpendix is there any reason in the future we can't add a check if the list is empty to the cardlist select and just return an empty list in that case? This can cause issues where a player is stuck without the ability to do anything because there are no valid checks if it isn't properly guarded against, which also causes dual checks like these (attackRequirement checking your hand and onAttack checking your hand)

Reply:
@flagrama that is a good idea. I was thinking to do it on the client level to affect all selections. A new business logic wrapping client class should handle auto selections, default values and so on. 
The second option is to do it on cardlist level as an simple 1 step fix. we already have the other use cases guarded by implementation logic anyway.